app/models/user.rb

```diff
@@ -50,6 +78,13 @@ def generate_unique_token
     retry
   end
 
+  def invalidate_token!
+    digest = reset_digest
+    update! reset_sent_at: nil, reset_digest: nil # Remove expired/valid tokens.
+    logger.info "User(id):#{id} reset password token with digest:#{digest} got removed."
```

Comment:
We definitely shouldn't be logging the digest, as it's considered senstiive data.

I would just `update! reset_sent_at: nil, reset_digest: nil` without any logging

Reply:
You're right, I wasn't thinking about it.
Will change it.